test/indexer/test_hca_indexer.py

```diff
@@ -698,6 +698,27 @@ def test_metadata_generator(self):
             self.assertEqual(uuid, metadata_row['bundle_uuid'])
             self.assertEqual(version, metadata_row['bundle_version'])
 
+    def test_exclude_elasticsearch_mappings(self):
```

Comment:
```diff
Index: test/indexer/test_hca_indexer.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- test/indexer/test_hca_indexer.py	(date 1558742563000)
+++ test/indexer/test_hca_indexer.py	(date 1558742551549)
@@ -698,26 +698,30 @@
             self.assertEqual(uuid, metadata_row['bundle_uuid'])
             self.assertEqual(version, metadata_row['bundle_version'])
 
-    def test_exclude_elasticsearch_mappings(self):
+    def test_metadata_field_exclusion(self):
         self._index_canned_bundle(self.old_bundle)
-        bundles_mapping = self.es_client.indices.get_mapping(index=config.es_index_name('bundles'))
-        metadata_mapping = bundles_mapping[config.es_index_name('bundles')]['mappings']['doc']['properties']['metadata']
+
+        # Check that the dynamic mapping has the field disabled
+        bundles_index = config.es_index_name('bundles')
+        mapping = self.es_client.indices.get_mapping(index=bundles_index)
+        self.assertFalse(mapping[bundles_index]['mappings']['doc']['properties']['metadata']['enabled'])
+
+        # Ensure that a metadata row exists …
         hits = self._get_all_hits()
-        bundles_hit = one([hit['_source'] for hit in hits if hit['_index'] == config.es_index_name('bundles')])
-
+        bundles_hit = one(hit['_source'] for hit in hits if hit['_index'] == bundles_index)
         for metadata_row in bundles_hit['metadata']:
             self.assertEqual(self.old_bundle, (metadata_row['bundle_uuid'], metadata_row['bundle_version']))
-        self.assertFalse(metadata_mapping['enabled'])
 
-        search_on_metadata = self.es_client.search(index=config.es_index_name('bundles'),
-                                                   body={
-                                                       "query": {
-                                                           "match": {
-                                                               "metadata.bundle_uuid": self.old_bundle[0]
-                                                           }
-                                                       }
-                                                   })
-        self.assertEqual(0, search_on_metadata["hits"]["total"])
+        # … but that it can't be used for queries
+        response = self.es_client.search(index=bundles_index,
+                                         body={
+                                             "query": {
+                                                 "match": {
+                                                     "metadata.bundle_uuid": self.old_bundle[0]
+                                                 }
+                                             }
+                                         })
+        self.assertEqual(0, response["hits"]["total"])
 
 
 class TestValidNotificationRequests(LocalAppTestCase):
```

Reply:
I would also add a search that proves that you *can* find the bundle contributions when you use the top level `bundle_uuid` field, not the one stashed away under `metadata[*].bundle_uuid`.